shell_sort/ShellSort.java

```diff
@@ -0,0 +1,41 @@
+public class ShellSort {
+	/**
+	* sorting function
+	* Worst case time complexity = O(n^2)
+	* Best case complexity = O(nlog(n))
+	* n is input size
+	*/
```

Comment:
As mentioned in the `.editorconfig`, space indentation should be used for java files.